cli/command/container/list.go

```diff
@@ -109,33 +108,17 @@ func buildContainerListOptions(opts *psOptions) (*types.ContainerListOptions, er
 func runPs(dockerCli command.Cli, options *psOptions) error {
 	ctx := context.Background()
 
+	if len(options.format) == 0 && !options.quiet {
+		options.format = dockerCli.ConfigFile().PsFormat
+	}
+
```

Comment:
```suggestion
	if len(options.format) == 0 && !options.quiet {
		if len(dockerCli.ConfigFile().PsFormat) > 0 {
			options.format = dockerCli.ConfigFile().PsFormat
		} else {
			options.format = formatter.TableFormatKey
		}
	}
```

I think we can merge our two if statements into one code block. We initially check whether a format had been specified or if the quiet option had been selected, then proceed to check the length of `PsFormat` within the config file to correctly assign a format.

What do you think about this?

Reply:
🤔 let me think (I think I had a reason)

Ah, so we _could_ probably move them into `buildContainerListOptions`. Looking at it, I guess 


The reason I'm not setting the format to `formatter.TableFormatKey` here (but doing that at the end) was for two reasons;

- I wanted the format to stay empty (if no format was given, and no `psFormat` is set) otherwise the template parsing code (for the size check) would be called, but not needed (overall, the code is a bit of a "hack")
- However, if no format is set at the end (even if `quiet` is set), we must set it to the default "table" value, to make this logic work; https://github.com/docker/cli/blob/d0df532a25ad18e8b4325edcf46f3d0f05b47708/cli/command/formatter/container.go#L30-L33


FWIW, I think we need to look at the logic in that code above as well (as a follow-up), to have that code pick the default (if format is empty), but also consider to make  `quiet`  work even if we would pass it a custom format.

Reply:
Ah! Right; and the reason I did the `options.format = dockerCli.ConfigFile().PsFormat` here, is that `buildContainerListOptions() didn't have access to `dockerCli` 😅

Reply:
I can add a comment to that line though to explain why the logic is the way it is; one moment.

Reply:
Ah that makes a lot of sense!